h/static/scripts/group-forms/components/ModerationStatusSelect.tsx

```diff
@@ -7,14 +7,36 @@ import {
   DottedCircleIcon,
   Select,
 } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import { Fragment } from 'preact';
 
 import type { ModerationStatus } from '../utils/api';
 import { moderationStatusToLabel } from '../utils/moderation-status';
 
 export type ModerationStatusSelectProps = {
-  selected?: ModerationStatus;
   onChange: (status?: ModerationStatus) => void;
-};
+  alignListbox?: 'right' | 'left';
+
+  /**
+   * `default`: Use same color palette regardless of selected status.
+   * `colored`: Use a color palette that represents selected status.
+   */
+  variant?: 'default' | 'colored';
+} & (
+  | {
+      /**
+       * Indicates this dropdown can skip the selected value, which will cause
+       * the value to fall back to "all" moderation statuses
+       */
+      valueRequired?: false;
```

Comment:
The effect of `valueRequired` is to switch the control between "select" and "filter" modes of operation. It affects not only whether a value is required, but also the icon and whether an "All" option is present. I would suggest to model this more explicitly with something like `mode: 'select' | 'filter'`. Alternatively you could create separate components which wrap a common base.

Reply:
Yeah, I was thinking on that precisely